src/blk/kernel/KernelDevice.cc

```diff
@@ -295,9 +295,13 @@ int KernelDevice::open(const string& p)
     goto out_fail;
   }
 
-  r = _aio_start();
-  if (r < 0) {
-    goto out_fail;
+  if (size > 0) {
```

Comment:
how device size can be zero? Maybe the code getting the device size is wrong?

The case failing for us is virtio-blk device in minikube (libvirt+qemu) and lima/vz (apple virtualization framework). The device size works in lsblk, no reason to have zero size.

Reply:
@nirs
From https://github.com/rook/rook/issues/15420#issuecomment-2682428598 
```
  -4> 2025-02-25T15:31:09.923+0000 7f6329c28c00 -1 bluestore(/dev/nbd0) No valid bdev label found
    -2> 2025-02-25T15:31:09.923+0000 7f6329c28c00 -1 bdev is_valid_io 0~1000 block_size 1000 size 0
    -1> 2025-02-25T15:31:09.923+0000 7f6329c28c00 -1 /home/jenkins-build/build/workspace/ceph-dev-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos9/DIST/centos9/MACHINE_SIZE/gigantic/release/19.3.0-7819-g290894cc/rpm/el9/BUILD/ceph-19.3.0-7819-g290894cc/src/blk/kernel/KernelDevice.cc: In function 'virtual int KernelDevice::aio_read(uint64_t, uint64_t, ceph::bufferlist*, IOContext*)' thread 7f6329c28c00 time 2025
-02-25T15:31:09.927527+0000
```
So obviously /dev/nbd0 had size 0.

Reply:
`bdev is_valid_io 0~1000 block_size 1000 size 0`
"size 0" means that device size is 0.